specs/transport/car/carv2/index.md

```diff
@@ -0,0 +1,177 @@
+---
+title: "CARv2 Specification"
+eleventyNavigation:
+  synopsys: "Specification for the CARv2 format"
+---
+
+# Specification: Content Addressable aRchives (CAR / .car) v2
+
+**Status: Draft**
+
+* [Summary](#summary)
+* [Format Description](#format-description)
+  * [Pragma](#pragma)
+  * [Header](#header)
+  * [Characteristics](#characteristics)
+  * [CARv1 data payload](#carv1-data-payload)
+  * [Index payload](#index-payload)
+  * [Index format](#index-format)
+    * [Format `0x300000`: IndexSorted](#format-0x300000-indexsorted)
+* [Implementations](#implementations)
+* [Test Fixtures](#test-fixtures)
+
+## Summary
+
+CARv2 is a minimal upgrade to the [CARv1](../carv1/index.md) format with the primary aim of adding an optional index within the format for fast random-access to blocks.
+
+CARv2 makes use of CARv1 by wrapping a properly formed CARv1 with a prefix containing a pragma and header, and a suffix containing the optional index data. Once the offset and length of the CARv1 bytes are determined using CARv2 parsing rules, an existing CARv1 decoder can be used to read the roots and `CID:Bytes` pairs. Likewise, a CARv1 encoder can be used to encode this data for wrapping by a CARv2 encoder.
```

Comment:
> Likewise, a CARv1 encoder can be used to encode this data for wrapping by a CARv2 encoder.

Thinking out loud: Though accurate, we probably want to encourage folks to use a CAR v2 encoder so that index can be generated as blocks are added. Otherwise, an archive produced by a CAR v1 encoder and passed to a CAR v2 for encoding would mean that blocks need to be re-scanned for the index to be generated.

Reply:
:thumbsup: I'll soften the language to make it more of a theoretical point:

>  Though not necessarily ideal, an existing CARv1 decoder could be used to read the roots and `CID:Bytes` pairs. Likewise, a CARv1 encoder could be be used to encode this data for wrapping by a CARv2 encoder as the payload is the same format.